ipaserver/install/replication.py

```diff
@@ -1077,13 +1078,10 @@ def setup_winsync_replication(self,
         self.ad_suffix = ""
         try:
             # Validate AD connection
-            ad_conn = ldap.initialize('ldap://%s' % ipautil.format_netloc(ad_dc_name))
-            # the next one is to workaround bugs arounf opendalp libs+NSS db
-            # we need to first specify the OPT_X_TLS_CACERTFILE and _after_
-            # that initialize the context to prevent TLS connection errors:
-            # https://bugzilla.redhat.com/show_bug.cgi?id=800787
-            ad_conn.set_option(ldap.OPT_X_TLS_CACERTFILE, cacert)
-            ad_conn.set_option(ldap.OPT_X_TLS_NEWCTX, 0)
+            ad_conn = ldap_initialize(
```

Comment:
This change modifies a behavior: we were ensuring no new context is used here but with ldap_initialize() we will now force OPT_X_TLS_NEWCTX if cacertfile is passed which is against what this code is intending.

Reply:
I don't understand what you are trying to tell me. The code already creates a new context in ``set_option(ldap.OPT_X_TLS_NEWCTX, 0)``. The value ``0`` does not mean "no new context". It actually creates a new context. The documentation https://linux.die.net/man/3/ldap_set_option is misleading. Any non-NULL value creates a new context.

> LDAP_OPT_X_TLS_NEWCTX
> Instructs the library to create a new TLS library context. invalue must be const int *. A non-zero value pointed to by invalue tells the library to create a context for a server. 

A new context **must** be created whenever a TLS related option is changed on a connection. Otherwise OpenLDAP will silently ignore any TLS settings. This is one of the subtle bugs I was referring to.